### PR TITLE
TPM1 extension and fixes

### PR DESCRIPTION
**Pull request by zaolin**
* Add ResetLock function
* Add Seal function
* Add Unseal function
* Fix parameters of ClearOwnership and TakeOwnership
* Set default constants
* Fix info structure parsing

### CODE REVIEW

File: pkg/tpm/tpm.go [hunk 1]
```diff
@@ -118,6 +134,16 @@ func getInfo() (*Info, error) {
 		return nil, err
 	}
 
+	manufacturePrefix := "Manufacturer: "
+	var manufactureID string
```
Comment (insomniacslk) [inline pkg/tpm/tpm.go]: typo: manufacture -> manufacturer

File: pkg/tpm/tpm.go [hunk 2]
```diff
@@ -118,6 +134,16 @@ func getInfo() (*Info, error) {
...
+	for _, lineBytes := range bytes.Split(caps, []byte{'\n'}) {
+		line := string(lineBytes)
+		if strings.HasPrefix(line, manufacturePrefix) {
+			manufactureID = line[len(manufacturePrefix):]
+		}
+	}
+	manufacture := Manufactures[manufactureID]
```
Comment (insomniacslk) [inline pkg/tpm/tpm.go]: this will return an empty string if the manufacturer ID is unknown. Do we want to explicitly signal that this is unknown? E.g. by setting it to the string "< unknown >" ? (spaces added around "unknown" or github would eat them)

File: pkg/tpm/tpm.go [hunk 3]
```diff
@@ -127,24 +153,25 @@ func getInfo() (*Info, error) {
 		}
 	}
 
-	owned, err := strconv.ParseBool(string(ownedBytes))
+	owned, err := strconv.ParseBool(string(ownedBytes[0]))
```
Comment (insomniacslk) [inline pkg/tpm/tpm.go]: was this failing because of the newline?

File: pkg/tpm/tpm1.go [hunk 4]
```diff
@@ -157,10 +180,11 @@ func (t *TPM1) Close() {
 // Summary returns a string with formatted TPM information
 func (t TPM1) Summary() string {
```
Comment (insomniacslk) [inline pkg/tpm/tpm1.go:181]: as discussed in an unrelated diff (in linuxboot/fiano), `Summary` is suboptimal and we'd better use JSON and (if necessary) introspection. I agree with that, and we can implement it in the next diff. Then we can remove Summary entirely

File: pkg/tpm/tpm.go (hunk 1 above)
Comment (zaolin) [reply pkg/tpm/tpm.go]: Done

File: pkg/tpm/tpm.go (hunk 2 above)
Comment (zaolin) [reply pkg/tpm/tpm.go]: Done

File: pkg/tpm/tpm.go (hunk 3 above)
Comment (zaolin) [reply pkg/tpm/tpm.go]: yes

Comment (insomniacslk) [reply pkg/tpm/tpm.go]: ParseBool accepts 0/false/FALSE, 1/true/TRUE. I don't expect the sysfs interface to return funny things in the future, but just for more robustness I'd rather strip off the newline, make sure we have only one line, and convert it to int. It's a common and non-trivial operation so I'd refactor it out to a small function

File: pkg/tpm/tpm.go [hunk 5]
```diff
@@ -46,22 +57,32 @@ type TPM interface {
...
 type Info struct {
+	Manufacturer           string
 	Specification          string
 	Owned                  bool
 	Active                 bool
 	Enabled                bool
 	TemporarilyDeactivated bool
 }
 
+func formatBoolFromSysfs(data []byte) string {
```
Comment (insomniacslk) [inline pkg/tpm/tpm.go]: We can simplify, make it like this:

```
func bytesToBool(data []byte) (bool, error) {
    s := strings.TrimSuffix(string(data), "\n")
    return strconv.ParseBool(s)
}
```

Then below just call it like
```
owned, err := bytesToBool(ownedBytes)
if err != nil {
    ...
}
```

**Review by insomniacslk — APPROVED** [review]
